### PR TITLE
Script: Migrate script functional tests to canton

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: daml-script/test/BUILD.bazel [hunk 1]
```diff
@@ -244,40 +241,6 @@ da_scala_test_suite(
     ],
 )
 
-# TODO: https://github.com/digital-asset/daml/issues/13737
-#  the test is flaky on windows.
```
Comment (cocreature) [inline daml-script/test/BUILD.bazel:248]: is it not flaky anymore?

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala [hunk 2]
```diff
@@ -428,8 +448,10 @@ abstract class AbstractFuncIT
         }
       }
     }
+    // TODO https://github.com/digital-asset/daml/issues/15882
```
Comment (cocreature) [inline daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala:451]: so sandbox on x supported consortium parties?

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala [hunk 3]
```diff
@@ -183,8 +193,18 @@ abstract class AbstractFuncIT
           )
         } yield {
           assert(vals.size == 2)
-          assert(vals.get(0) == SParty(Party.assertFromString("carol")))
-          assert(vals.get(1) == SParty(Party.assertFromString("dan")))
+          inside(vals.get(0)) { case SParty(partyId) =>
+            inside(partyId.split("::")) { case Array(prefix, suffix) =>
+              prefix shouldBe "carol"
+              suffix.length shouldBe 68
```
Comment (cocreature) [inline daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala:199]: not sure this assertion is super useful.

**Review by cocreature — APPROVED** [review]
Nice work thanks!

File: daml-script/test/BUILD.bazel (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-script/test/BUILD.bazel:248]: I forget to drop the comment in #16489

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala:451]: I am not completely sure.

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala:199]: We should review the test.

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala:451]: It is failing with Canton, taking into account @nickchapman-da is refactoring current implementation, I do not think is worth to spend time with it.

Comment (cocreature) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractFuncIT.scala:451]: wfm, I was just surprised that sandbox on x even supported it.

File: build.ps1 [hunk 4]
```diff
@@ -73,10 +73,11 @@ if ($env:SKIP_TESTS -ceq "False") {
     ./ci/remap-scala-test-short-names.ps1 `
       | Out-File -Encoding UTF8 -NoNewline scala-test-suite-name-map.json
 
-    bazel test //... `
+    bazel test //daml-script/test:test_test_suite_src_com_digitalasset_daml_lf_engine_script_test_FuncWallClockIT.scala //daml-script/test:test_test_suite_src_com_digitalasset_daml_lf_engine_script_test_FuncStaticTimeIT.scala `
```
Comment (remyhaemmerle-da) [inline build.ps1:76]: Temporary to check the test is not flaky on CI.

**Pull request merged**